service/build.go

```diff
@@ -414,22 +409,24 @@ func (uis *UIServer) buildHistory(w http.ResponseWriter, r *http.Request) {
 	gimlet.WriteJSON(w, history)
 }
 
-func getBuildTaskMap(builds []build.Build) (map[string]map[string]task.Task, error) {
+// getBuildTaskMaps returns
+// 1. a map of build ID to a task ID slice
+// 2. a map of task IDs to task documents
+func getBuildTaskMaps(builds []build.Build) (map[string][]string, map[string]task.Task, error) {
 	buildIds := make([]string, 0, len(builds))
 	for _, b := range builds {
 		buildIds = append(buildIds, b.Id)
 	}
 	tasksForBuilds, err := task.FindAll(task.ByBuildIds(buildIds).WithFields(task.BuildIdKey, task.DisplayNameKey, task.StatusKey, task.DetailsKey))
 	if err != nil {
-		return nil, errors.Wrap(err, "can't get tasks for builds")
+		return nil, nil, errors.Wrap(err, "can't get tasks for builds")
 	}
-	buildTaskMap := make(map[string]map[string]task.Task)
+	buildTaskIDsMap := make(map[string][]string)
```

Comment:
Actually follow up question: do we even need two maps? Could we do map[string][]task.Task{}? We aren't checking if the task already exists in the list at 427 anyway, I'm not sure we need the taskIds separately.

Reply:
You're right. What I really intended to do was to look up the tasks from the task cache.
I've fixed it. 🤦 
Now it returns a single map from task id to task.